frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/initial-state/storage-tab-initial-state.ts

```diff
@@ -60,42 +59,47 @@ export const windowsToolsStorage: VMWizardStorage = {
   }).asResource(),
 };
 
-const getUrlStorage = (storageClassConfigMap: ConfigMapKind) => ({
-  type: VMWizardStorageType.PROVISION_SOURCE_DISK,
-  disk: DiskWrapper.initializeFromSimpleData({
-    name: ROOT_DISK_NAME,
-    type: DiskType.DISK,
-    bus: DiskBus.VIRTIO,
-    bootOrder: 1,
-  }).asResource(),
-  volume: VolumeWrapper.initializeFromSimpleData({
-    name: ROOT_DISK_NAME,
-    type: VolumeType.DATA_VOLUME,
-    typeData: { name: prefixedID(VM_TEMPLATE_NAME_PARAMETER, ROOT_DISK_NAME) },
-  }).asResource(),
-  dataVolume: DataVolumeWrapper.initializeFromSimpleData({
-    name: prefixedID(VM_TEMPLATE_NAME_PARAMETER, ROOT_DISK_NAME),
-    type: DataVolumeSourceType.HTTP,
-    typeData: { url: '' },
-    size: 10,
-    unit: BinaryUnit.Gi,
-  })
-    .setVolumeMode(getDefaultSCVolumeMode(storageClassConfigMap))
-    .setAccessModes(getDefaultSCAccessModes(storageClassConfigMap))
-    .asResource(),
-  editConfig: {
-    isFieldEditableOverride: {
-      source: false,
+const getUrlStorage = (vmName: string, storageClassConfigMap: ConfigMapKind) => {
+  const dataVolumeName = generateDataVolumeName(vmName, ROOT_DISK_NAME);
```

Comment:
the vmName might change meanwhile here as well, so I guess it is not so useful to pass it. Let's make it easier to understand and just use `'vm'` constant instead

Reply:
Done!